PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -266,7 +266,7 @@ def nanoAOD_runMETfixEE2017(process,isData):
     process.nanoSequenceCommon.insert(process.nanoSequenceCommon.index(jetSequence),process.fullPatMetSequenceFixEE2017)
 
 def nanoAOD_customizeCommon(process):
-#    makePuppiesFromMiniAOD(process,True) # call this here as it calls switchOnVIDPhotonIdProducer
+    makePuppiesFromMiniAOD(process,True) # call this here as it calls switchOnVIDPhotonIdProducer
```

Comment:
This should be put beyond a process modifier since V14 Puppi is available already in the re-miniAOD

for master:
if run2_nanoAOD_106Xv1 (+ any other old):
        makePuppiesFromMiniAOD(process,True) 

for 10_6:
if NOT run2_miniAOD_devel:
        makePuppiesFromMiniAOD(process,True)

Reply:
You need to decide which tune you want for
run2_miniAOD_80XLegacy, run2_nanoAOD_94X2016, run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_102Xv1, run2_nanoAOD_106Xv1

This need to be in compatible with the JEC analyst have in the GT, unless you plan to reserve the previous.

Reply:
Hello,
We want the old PUPPI tune for all the mentioned ones, except for run2_nanoAOD_106Xv1

Reply:
Hello, I am concerned about the code crashing if the makePuppiesFromMiniAOD(process,True) is not called at all under one of the conditions. It seems to be required to obtain the puppiMETSequence 
and is recommended to be included here:
https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#Puppi_MET
even if the PUPPI tune is not to be updated.

I tried testing out the code without that the call to makePuppiesFromMiniAOD(process,True), and the code crashes with the message
'Process' object has no attribute 'puppiMETSequence'

Reply:
Yes so makePuppiesFromMiniAOD(process,True) is apparently needed. 
Turning on/off the new PUPPI tune is done via:
process.puppiNoLep.useExistingWeights = False
process.puppi.useExistingWeights = False
This is where the modifiers should be placed from what I understand.

Reply:
I have set these are explicitly True now, to use the existing weights, that should resolve the concern.

Reply:
This line is a general call, for all jetMET objects and not only MET. Why you want always ignore what is stored in miniAOD and need to recompute the puppiWeigth on the fly always  ?

I expected this like actived only for the old-miniAOD:

for era in [run2_miniAOD_80XLegacy, run2_nanoAOD_92X, run2_nanoAOD_94XMiniAODv1, \
             run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_94X2016, run2_nanoAOD_102Xv1, \
             run2_nanoAOD_106Xv1]:
makePuppiesFromMiniAOD(process,True)

In master you have the latest tune and you can take the existing Puppi weights already and the JEC in synch with those.
For the older version you have the V11 tune for which you do not have and will not have proper JEC, so here we need to recompute and get in synch with the tune in the release.

Reply:
This call is not to ignore what is stored in miniAOD, 
I have explicitly set 
process.puppiNoLep.useExistingWeights = True
process.puppi.useExistingWeights = True
so that the existing PUPPI weights from miniAOD are used.
This call is there to obtain the puppiMETSequence without which the code for re-applying the JECs for PUPPI MET crashes.
It is recommended to be included here:
https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#Puppi_MET
even if the PUPPI tune is not to be updated.

Reply:
After some discussion with Maria, we believe the two following lines:
process.puppiNoLep.useExistingWeights = True
process.puppi.useExistingWeights = True
Then an era modifier should be used to set this to false when needed (only for run2_nanoAOD_106Xv1 in 106X from what I understand).

Reply:
yes correct

I suggest to put `useExistingWeights = False` for EOY and UL106 basically for all these
run2_nanoAOD_92X, run2_nanoAOD_94XMiniAODv1, 
run2_nanoAOD_94XMiniAODv2, run2_nanoAOD_94X2016, run2_nanoAOD_102Xv1, 
run2_nanoAOD_106Xv1

Reply:
Do I understand correctly that for the version to be merged with the master, no era modifier is required?
Then I'll just move those two lines to the nanoAOD_customizeCommon function for the version to be merged with the master, and add the era modifier later for the 106X version.

Reply:
Okay, I will make the changes and update

Reply:
Hello, updated with the latest commit, please check